pkg/apis/apps/validation/validation.go

```diff
@@ -150,6 +159,7 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path, op
 	}
 
 	allErrs = append(allErrs, ValidatePersistentVolumeClaimRetentionPolicy(spec.PersistentVolumeClaimRetentionPolicy, fldPath.Child("persistentVolumeClaimRetentionPolicy"))...)
+	allErrs = append(allErrs, ValidateVolumeClaimTemplates(spec.VolumeClaimTemplates, fldPath.Child("volumeClaimTemplates"))...)
```

Comment:
this has to be conditional based on an option in setOpts so we don't break update of existing SS objects with already-invalid data in this field (e.g. `if !setOpts.SkipValidateVolumeClaimTemplates bool`, see above diff for an example)

Add unit tests for tolerating updates to a statefulset that has invalid data in the old object already